File: backend/app/domain/dating_profiles/enums.py
from enum import Enum


class DatingStatus(Enum):
    OPEN = "open"
    BREAK = "break"
    WINGING = "winging"


class City(Enum):
    BOSTON = "Boston"
    NEW_YORK = "New York"


class Religion(Enum):
    MUSLIM = "Muslim"
    CHRISTIAN = "Christian"
    JEWISH = "Jewish"
    HINDU = "Hindu"
    BUDDHIST = "Buddhist"
    SIKH = "Sikh"
    AGNOSTIC = "Agnostic"
    ATHEIST = "Atheist"
    OTHER = "Other"
    PREFER_NOT_TO_SAY = "Prefer not to say"


class Interest(Enum):
    TRAVEL = "Travel"
    FITNESS = "Fitness"
    COOKING = "Cooking"
    MUSIC = "Music"
    ART = "Art"
    MOVIES = "Movies"
    BOOKS = "Books"
    GAMING = "Gaming"
    OUTDOORS = "Outdoors"
    SPORTS = "Sports"
    TECHNOLOGY = "Technology"
    FASHION = "Fashion"
    FOOD = "Food"
    PHOTOGRAPHY = "Photography"
    DANCE = "Dance"
    VOLUNTEERING = "Volunteering"

File: app/constants/enums.ts
import type { City, Gender, Religion, Interest } from '@/lib/api/generated/model';

export const CITIES = ['Boston', 'New York'] as const satisfies City[];

export const GENDERS = ['Male', 'Female', 'Non-Binary'] as const satisfies Gender[];

export const RELIGIONS = [
  'Muslim',
  'Christian',
  'Jewish',
  'Hindu',
  'Buddhist',
  'Sikh',
  'Agnostic',
  'Atheist',
  'Other',
  'Prefer not to say',
] as const satisfies Religion[];

export const INTERESTS = [
  'Travel',
  'Fitness',
  'Cooking',
  'Music',
  'Art',
  'Movies',
  'Books',
  'Gaming',
  'Outdoors',
  'Sports',
  'Technology',
  'Fashion',
  'Food',
  'Photography',
  'Dance',
  'Volunteering',
] as const satisfies Interest[];
